src/reflection/reflection.jl

```diff
@@ -89,7 +89,13 @@ function invoke_tweaks!(ci::CodeInfo)
   ci.slotflags = [0x00, ci.slotflags[1], 0x00, ci.slotflags[2:end]...]
   ci.code = map(ci.code) do x
     prewalk(x) do x
-      x isa SlotNumber ? SlotNumber(x.id == 1 ? 2 : x.id+2) : x
+      if x isa SlotNumber
+        SlotNumber(x.id == 1 ? 2 : x.id+2)
+      elseif x isa Core.ReturnNode && x.val isa SlotNumber
```

Comment:
```suggestion
      elseif x isa Core.ReturnNode && isdefined(x, :val) && x.val isa SlotNumber
```

Reply:
good catch! Are there other special nodes that could break this bit?

Reply:
we could take inspiration from this other prewalk use over instructions so `GotoIfNot`:

https://github.com/FluxML/IRTools.jl/blob/dd1f2c212258001ea565df696841929ad0fcb614/src/reflection/utils.jl#L66-L70